roombasim/ai/task.py

```diff
@@ -28,14 +28,39 @@ def __init__(self, tasks):
         self.tasks['idle'] = IdleTask
         self.switch_task('idle')
 
-    def switch_task(self, task_name, **params):
+    def switch_task(self, task_name, callback=None, **params):
         '''
         Switch to a new task by name with optional params.
 
+        If the callback parameter is specified, once the task
+        completes, it will be called with the following signature:
+
+        >>> callback(status, message)
+
         Note: params are supplied as kwargs and will be given
         to the task constructor as kwargs.
         '''
         self.current = self.tasks[task_name](**params)
+        self.current.set_completion_callback(self.callback_wrapper(callback))
+
+    def callback_wrapper(self, callback):
+        '''
+        Creates a callback function of signature (status, method)
+        that will set the current task to 'idle' and will optionally
+        call an additional callback function if it exists.
+        '''
+        if callback != None:
```

Comment:
Isn't it better to say `if callback is not None:` ?

Reply:
Technically yes, because `None` is a singleton

Reply:
fixed